app/src/main/java/espertolabs/esperto_ble_watch/LoginActivity.java

```diff
@@ -164,12 +164,14 @@ public void run(){
                     if(response != ""){
                         Gson g = new Gson();
                         UserAccount user = g.fromJson(response, UserAccount.class);
-                        Log.d("built_user", user.getName());
-
-                        Intent displaySummary = new Intent(getApplicationContext(), SummaryActivity.class);
-                        displaySummary.putExtra("deviceAddress", "D4:49:8C:44:48:82");
-                        displaySummary.putExtra("user_obj", user);
-                        startActivity(displaySummary);
+                        if(user != null){
+                            Log.d("built_user", user.getName());
+
+                            Intent displaySummary = new Intent(getApplicationContext(), SummaryActivity.class);
+                            displaySummary.putExtra("deviceAddress", "D4:49:8C:44:48:82");
```

Comment:
Should deviceAddress be a part of the user object and fetched from the server?

Reply:
yeah it should be, thanks